File: Data Structures & Algorithms/two-integer-sum/submission-11.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i, num in enumerate(nums):
            operand = target - num
            if operand in map:
                return [map[operand], i]
            map[num] = i
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-11.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numSet = new HashSet<>();
        for(int num : nums){
            numSet.add(num);
        }
        int max = 0;
        for(int num : nums){
            int curr = num;
            int longest  = 1;
            if(!numSet.contains(curr - 1)){
                while(numSet.contains(curr + 1)){
                    longest++;
                    curr++;
                }
            }
            max = Math.max(longest, max);
        }
        return max;
    }
}
